universal-application-tool-0.0.1/app/services/cloud/azure/BlobStorage.java

```diff
@@ -200,23 +200,23 @@ public String getBlobUrl(String fileName) {
           checkNotNull(config).getString(AZURE_LOCAL_CONNECTION_STRING_CONF_PATH);
       this.blobServiceClient =
           new BlobServiceClientBuilder().connectionString(connectionString).buildClient();
+      String baseUrl = checkNotNull(config).getString("base_url");
 
-      setCorsRules(blobServiceClient);
+      setCorsRules(blobServiceClient, baseUrl);
       this.blobContainerClient = blobServiceClient.getBlobContainerClient(container);
       if (!blobContainerClient.exists()) {
         blobContainerClient.create();
       }
     }
 
-    private void setCorsRules(BlobServiceClient blobServiceClient) {
+    private void setCorsRules(BlobServiceClient blobServiceClient, String baseUrl) {
       BlobServiceProperties properties =
           new BlobServiceProperties()
               .setCors(
                   List.of(
                       new BlobCorsRule()
-                          .setAllowedOrigins("*")
-                          .setAllowedHeaders("*")
-                          .setExposedHeaders("*")
+                          .setAllowedOrigins(baseUrl) 
+                          .setAllowedHeaders("content-type,x-ms-blob-type,x-ms-client-request-id,x-ms-version")
```

Comment:
nit: please add a comment describing why these specific headers and a link to Azure documentation if appropriate/available

Reply:
also done!